hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,548 @@
+On the other hand in the faster Dual- or Quad-modes, all data lines are bidirectional, and in Quad mode the number of data lines increases to four.
+For the purposes of this IP, Dual or Quad-mode commands can be thought of as consisting of up to four command phases in which the host:
+1. Transmits instructions or data at single-line rate,
+2. Transmits instructions address or data on 2 or 4 data lines,
+3. Holds the bus in a high-impedance state for some number of "dummy" clock cycles (neither side transmits), or
+4. Receives information from the target device.
+
+This four phase command scheme is also applied to half-duplex interactions when in standard mode.
+
+{{< wavejson >}}
+{signal: [
+  {name: "clk_core_i", wave: "p.............................."},
+  {name: "sck (CPOL=0)", wave: "0.1010101010101010101010101010."},
+  {name: "cs_n", wave: "10............................1"},
+  {name: "sd[0]", wave: "22.2.2.2.2.2.2.2.2.2.z...2.2.x.", data: ["","cmd7", "cmd6", "cmd5", "cmd4", "cmd3", "cmd2", "cmd1", "cmd0",
+                                                             "out7", "out3", "in7", "in3" ]},
+  {name: "sd[1]", wave: "z................2.2.z...2.2.x.", data: ["out6", "out2", "in6", "in2"]},
+  {name: "sd[2]", wave: "z................2.2.z...2.2.x.", data: ["out5", "out1", "in5", "in1"]},
+  {name: "sd[3]", wave: "z................2.2.z...2.2.x.", data: ["out4", "out0", "in4", "in0"]},
+  {name: "Transaction phase:", wave: "x2...............3...4...5...x.", data: ['TX1', 'TXn', 'Dummy','RX'] },
+  {name: "Sampling Event (FULLCYC=0)", wave: "1.........................H.H.."},
+  {name: "Sampling Event (FULLCYC=1)", wave: "1..........................H.H."},
+],
+  head: {
+  text: "Example Quad SPI transaction: 1 byte TX (Single), 1 byte (Quad), 2 dummy cycles and 1 RX byte with CPHA=0"
+  },
+}
+{{< /wavejson >}}
+
+For even faster transfer rates, some flash chips support double transfer rate (DTR) variations to the SPI protocol wherein the device receives and transmits fresh data on *both* the leading and trailing edge.
+This IP only supports single transfer rate (STR), *not* DTR.
+A preliminary investigation of DTR transfer mode suggests that proper support for setup and hold times in this mode may require a level of sub-cycle timing control which is not currently planned for this IP.
+
+# Theory of Operations
+
+**TODO**: Review text for accuracy after review updates.
+
+## SPI_HOST IP Command Interface
+
+Issuing a command generally consists of the following steps:
+1. Configure the IP to be compatible with the target peripheral.
+The {{< regref "CONFIG_0" >}} multi-register has one set of configuration settings for each `cs_n` line.
+2. Load the instructions and data to be transmitted into the TXFIFO via the {{< regref "TXDATA" >}} register,
+3. Issue a 32-bit command to the appropriate Command multi-register (e.g. for controlling the device on `cs_n[0]` use {{< regref "COMMAND_0" >}}).
```

Comment:
Nit: I understand why you've not linked "Command" here, but maybe it would be better as COMMAND (or even `COMMAND`). At the moment, the capital looks a bit odd.